src/lib/dnssd/minimal_mdns/Server.cpp

```diff
@@ -209,73 +265,50 @@ CHIP_ERROR ServerBase::DirectSend(chip::System::PacketBufferHandle && data, cons
             continue;
         }
 
-        chip::Inet::InterfaceId boundIf = info->udp->GetBoundInterface();
+        chip::Inet::InterfaceId boundIf = info->listen_udp->GetBoundInterface();
 
         if ((boundIf.IsPresent()) && (boundIf != interface))
         {
             continue;
         }
 
-        return info->udp->SendTo(addr, port, std::move(data));
+        return info->listen_udp->SendTo(addr, port, std::move(data));
     }
 
     return CHIP_ERROR_NOT_CONNECTED;
 }
 
-CHIP_ERROR ServerBase::BroadcastSend(chip::System::PacketBufferHandle && data, uint16_t port, chip::Inet::InterfaceId interface,
-                                     chip::Inet::IPAddressType addressType)
+CHIP_ERROR ServerBase::BroadcastUnicastQuery(chip::System::PacketBufferHandle && data, uint16_t port)
 {
-    for (size_t i = 0; i < mEndpointCount; i++)
-    {
-        EndpointInfo * info = &mEndpoints[i];
-
-        if (info->udp == nullptr)
-        {
-            continue;
-        }
-
-        if ((info->interfaceId != interface) && (info->interfaceId != chip::Inet::InterfaceId::Null()))
-        {
-            continue;
-        }
-
-        if ((addressType != chip::Inet::IPAddressType::kAny) && (info->addressType != addressType))
-        {
-            continue;
-        }
+    QuerySocketDelegate delegate;
+    return BroadcastImpl(std::move(data), port, &delegate);
+}
 
-        CHIP_ERROR err;
+CHIP_ERROR ServerBase::BroadcastUnicastQuery(chip::System::PacketBufferHandle && data, uint16_t port,
```

Comment:
Rename to `BroadcastLegacyUnicastQuery` since that's what we do (not originating from 5353)

Reply:
I am thinking to leave it alone since a followup PR should make the path  configurable. That way we can have this do 'legacy' on linux/mac/android (or wherever conflicts can occur in receiving unicast answers) but most embedded platforms should not pay the cost of the unicast and legacy querying.

Reply:
I did add comments (but with the followup PR in mind, they say "MAY" for sending unicast packets from an ephemeral port.